### ISSUE TITLE
adapter-auto unable to detect production environment when deploying to Vercel

### ISSUE THREAD

**Issue by josefaidt** [sveltejs/kit#2956]
### Describe the bug

I started a new SvelteKit project and noticed it came with `@sveltejs/adapter-auto`. Upon attempting to deploy to Vercel I received an error regarding a missing build output to `public/`.

NOTE: this occurs after the successful build

```
> Using @sveltejs/adapter-auto
--
Could not detect a supported production environment. See https://kit.svelte.dev/docs#adapters to learn how to configure your app to run on the platform of your choosing
✔ done
Done in 1.54s.
Error: No Output Directory named "public" found after the Build completed. You can configure the Output Directory in your Project Settings. Learn More: https://vercel.link/missing-public-directory
```

Looking at the adapters.js file, ~is there an intended `auto` adapter for local builds (static/node)?~ just noticed the docs were updated to show removing the auto adapter in favor of node or static https://kit.svelte.dev/docs#adapters-building-a-node-app

```js
export const adapters = [
	{
		name: 'Vercel',
		test: () => !!process.env.VERCEL,
		module: '@sveltejs/adapter-vercel'
	},
	{
		name: 'Cloudflare Pages',
		test: () => !!process.env.CF_PAGES,
		module: '@sveltejs/adapter-cloudflare'
	},
	{
		name: 'Netlify',
		test: () => !!process.env.NETLIFY,
		module: '@sveltejs/adapter-netlify'
	}
];
```

### Reproduction

1. `npm init svelte@next my-app`
2. deploy skeleton project to Vercel with `vercel`

### System Info

```shell
System:
    OS: macOS 11.5.1
    CPU: (4) x64 Intel(R) Core(TM) i7-7567U CPU @ 3.50GHz
    Memory: 144.43 MB / 16.00 GB
    Shell: 3.3.1 - /usr/local/bin/fish
  Binaries:
    Node: 16.13.0 - /var/folders/pp/ty2pnrf50xg0zgv3033j83_m0000gn/T/fnm_multishells/38331_1638326972687/bin/node
    Yarn: 1.23.0-20210726.1745 - /var/folders/pp/ty2pnrf50xg0zgv3033j83_m0000gn/T/fnm_multishells/38331_1638326972687/bin/yarn
    npm: 8.1.0 - /var/folders/pp/ty2pnrf50xg0zgv3033j83_m0000gn/T/fnm_multishells/38331_1638326972687/bin/npm
  Browsers:
    Brave Browser: 92.1.28.105
    Chrome: 96.0.4664.55
    Chrome Canary: 98.0.4740.0
    Edge: 96.0.1054.41
    Firefox: 94.0.1
    Firefox Developer Edition: 68.0
    Safari: 14.1.2
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.3 
    @sveltejs/kit: next => 1.0.0-next.201 
    svelte: ^3.44.0 => 3.44.2 
```

### Severity

serious, but I can work around it

### Additional Information

It seems I can work around this by using the Vercel or static adapters directly.

**Comment by simonjcarr** [comment]
I get the same error. using both the demo app and a skeleton with some custom code. Preview works fine, but if try to dockerize the project and I get the following errors when I run the container

```
(node:1) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
/app/build/app.js:1
import { respond } from '@sveltejs/kit/ssr';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (node:internal/modules/cjs/loader:999:16)
    at Module._compile (node:internal/modules/cjs/loader:1047:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1112:10)
    at Module.load (node:internal/modules/cjs/loader:948:32)
    at Function.Module._load (node:internal/modules/cjs/loader:789:14)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:72:12)
    at node:internal/main/run_main_module:17:47
```

**Comment by aolose** [comment]
> I get the same error. using both the demo app and a skeleton with some custom code. Preview works fine, but if try to dockerize the project and I get the following errors when I run the container
> 
> ```
> (node:1) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
> (Use `node --trace-warnings ...` to show where the warning was created)
> /app/build/app.js:1
> import { respond } from '@sveltejs/kit/ssr';
> ^^^^^^
> 
> SyntaxError: Cannot use import statement outside a module
>     at wrapSafe (node:internal/modules/cjs/loader:999:16)
>     at Module._compile (node:internal/modules/cjs/loader:1047:27)
>     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1112:10)
>     at Module.load (node:internal/modules/cjs/loader:948:32)
>     at Function.Module._load (node:internal/modules/cjs/loader:789:14)
>     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:72:12)
>     at node:internal/main/run_main_module:17:47
> ```

I got the same error `Cannot use import statement outside a module`.
After I created the package.json with field `"type": "module"` in the build folder, the problem was solved.

you can try:
```
echo `{"type": "module"}` > xxx/build/package.json
```

**Comment by JLAcostaEC** [comment]
This Issue has some time, however...

I had the same problem and considering that the AdapterJS tests the environment variable, I was wondering if manually putting the environment variables in vercel removes the .env.VERCEL variable.

Effectively! if you manually set environment variables in Vercel remember to also set VERCEL=1 so that the svelteKit adapter detects that it is in Vercel.

**Comment by geoffrich** [comment]
I can't reproduce this issue. I initialized the demo app using `npm init` and deployed to Vercel with the default settings (which uses adapter-auto). Everything worked without any issue: https://sveltekit-demo-lime.vercel.app/

I also manually set an environment variable. The "automatically expose system environment variables" checkbox was enabled by default, so that should work out of the box as well.

![image](https://user-images.githubusercontent.com/4992896/156804864-4f967b38-09ba-494c-a7f4-b518f9ead666.png)
